### ISSUE TITLE
Terraform crash.log

### ISSUE THREAD

**Issue by cornevandyk** [hashicorp/terraform#10109]
Hallo, I just had a Terraform complete crash. Please let me know if you need additional information.

regards
Corne

### Terraform Version
0.7.9

### Debug Output
https://gist.github.com/cornevandyk/8356b2a563e6619c095764420a866d55

### Panic Output
https://gist.github.com/cornevandyk/8356b2a563e6619c095764420a866d55

**Comment by mitchellh** [comment]
(Notes)

Crash is in `data_aws_acm_certificate`, key part:

```
2016/11/14 16:59:50 [DEBUG] plugin: terraform: panic: interface conversion: interface is []interface {}, not []string
2016/11/14 16:59:50 [DEBUG] plugin: terraform: 
2016/11/14 16:59:50 [DEBUG] plugin: terraform: goroutine 138 [running]:
2016/11/14 16:59:50 [DEBUG] plugin: terraform: panic(0x267a5e0, 0xc42046cec0)
2016/11/14 16:59:50 [DEBUG] plugin: terraform: 	/opt/go/src/runtime/panic.go:500 +0x1a1
2016/11/14 16:59:50 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/builtin/providers/aws.dataSourceAwsAcmCertificateRead(0xc420264120, 0x2344c00, 0xc4204c89c0, 0x3f00000000000007, 0x0)
2016/11/14 16:59:50 [DEBUG] plugin: terraform: 	/opt/gopath/src/github.com/hashicorp/terraform/builtin/providers/aws/data_source_aws_acm_certificate.go:43 +0x177
2016/11/14 16:59:50 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/helper/schema.(*Resource).ReadDataApply(0xc4205ae4e0, 0xc4206a3dc0, 0x2344c00, 0xc4204c89c0, 0xc420276408, 0x1, 0x0)
2016/11/14 16:59:50 [DEBUG] plugin: terraform: 	/opt/gopath/src/github.com/hashicorp/terraform/helper/schema/resource.go:207 +0xda
2016/11/14 16:59:50 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/helper/schema.(*Provider).ReadDataApply(0xc4205441e0, 0xc4205f88c0, 0xc4206a3dc0, 0x0, 0x18, 0x18)
2016/11/14 16:59:50 [DEBUG] plugin: terraform: 	/opt/gopath/src/github.com/hashicorp/terraform/helper/schema/provider.go:317 +0x91
2016/11/14 16:59:50 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/plugin.(*ResourceProviderServer).ReadDataApply(0xc42006ee80, 0xc42069aec0, 0xc42069b220, 0x0, 0x0)
2016/11/14 16:59:50 [DEBUG] plugin: terraform: 	/opt/gopath/src/github.com/hashicorp/terraform/plugin/resource_provider.go:537 +0x4e
2016/11/14 16:59:50 [DEBUG] plugin: terraform: reflect.Value.call(0xc42046e6c0, 0xc420084130, 0x13, 0x2bda80e, 0x4, 0xc4204ffee0, 0x3, 0x3, 0x2, 0xc4203a1b40, ...)
2016/11/14 16:59:50 [DEBUG] plugin: terraform: 	/opt/go/src/reflect/value.go:434 +0x5c8
2016/11/14 16:59:50 [DEBUG] plugin: terraform: reflect.Value.Call(0xc42046e6c0, 0xc420084130, 0x13, 0xc4204ffee0, 0x3, 0x3, 0x0, 0xc420269f80, 0xa93830)
2016/11/14 16:59:50 [DEBUG] plugin: terraform: 	/opt/go/src/reflect/value.go:302 +0xa4
2016/11/14 16:59:50 [DEBUG] plugin: terraform: net/rpc.(*service).call(0xc4202a2b80, 0xc4202a2b40, 0xc420540b08, 0xc4200a0580, 0xc4205ec0e0, 0x2346d80, 0xc42069aec0, 0x16, 0x2346dc0, 0xc42069b220, ...)
2016/11/14 16:59:50 [DEBUG] plugin: terraform: 	/opt/go/src/net/rpc/server.go:383 +0x148
2016/11/14 16:59:50 [DEBUG] plugin: terraform: created by net/rpc.(*Server).ServeCodec
2016/11/14 16:59:50 [DEBUG] plugin: terraform: 	/opt/go/src/net/rpc/server.go:477 +0x421
```

**Comment by cornevandyk** [comment]
thanks Mitchell, I can remove this component for now. It is perhaps worth adding that my IDE (Pycharm using the HCL plugin) doesn't automatically suggest the `aws_acm_certificate` object that has been defined. Perhaps it doesn't like the wildcard domain that the certificate has been issued for?
The object is defined as
```data "aws_acm_certificate" "star-XXX-YYY" {
  domain = "*.XXX.YYY"
  statuses = ["ISSUED"]
}```

regards
Corne

**Comment by mrwacky42** [comment]
I also have a crash. Also a wildcard cert.

Seemingly the same panic:
```
2016/11/15 16:59:34 [DEBUG] plugin: terraform: panic: interface conversion: interface is []interface {}, not []string
2016/11/15 16:59:34 [DEBUG] plugin: terraform: 
2016/11/15 16:59:34 [DEBUG] plugin: terraform: goroutine 28 [running]:
2016/11/15 16:59:34 [DEBUG] plugin: terraform: panic(0x2698420, 0xc42061f340)
2016/11/15 16:59:34 [DEBUG] plugin: terraform: 	/usr/local/Cellar/go/1.7.3/libexec/src/runtime/panic.go:500 +0x1a1
2016/11/15 16:59:34 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/builtin/providers/aws.dataSourceAwsAcmCertificateRead(0xc42033a3c0, 0x2362200, 0xc4203b6d00, 0x4900000000000028, 0x0)
2016/11/15 16:59:34 [DEBUG] plugin: terraform: 	/private/tmp/terraform-20161109-45340-o0vw8d/terraform-0.7.10/src/github.com/hashicorp/terraform/builtin/providers/aws/data_source_aws_acm_certificate.go:69 +0x6e6
2016/11/15 16:59:34 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/helper/schema.(*Resource).ReadDataApply(0xc42050d3e0, 0xc420650680, 0x2362200, 0xc4203b6d00, 0xc4203acd78, 0x1, 0x0)
2016/11/15 16:59:34 [DEBUG] plugin: terraform: 	/private/tmp/terraform-20161109-45340-o0vw8d/terraform-0.7.10/src/github.com/hashicorp/terraform/helper/schema/resource.go:207 +0xda
2016/11/15 16:59:34 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/helper/schema.(*Provider).ReadDataApply(0xc42029b320, 0xc42064a050, 0xc420650680, 0x0, 0x18, 0x18)
2016/11/15 16:59:34 [DEBUG] plugin: terraform: 	/private/tmp/terraform-20161109-45340-o0vw8d/terraform-0.7.10/src/github.com/hashicorp/terraform/helper/schema/provider.go:317 +0x91
2016/11/15 16:59:34 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/plugin.(*ResourceProviderServer).ReadDataApply(0xc4200c5f40, 0xc42070e9a0, 0xc42070ec30, 0x0, 0x0)
2016/11/15 16:59:34 [DEBUG] plugin: terraform: 	/private/tmp/terraform-20161109-45340-o0vw8d/terraform-0.7.10/src/github.com/hashicorp/terraform/plugin/resource_provider.go:537 +0x4e
2016/11/15 16:59:34 [DEBUG] plugin: terraform: reflect.Value.call(0xc42045b5c0, 0xc4201b4670, 0x13, 0x2bfa312, 0x4, 0xc420169ed0, 0x3, 0x3, 0x0, 0x0, ...)
2016/11/15 16:59:34 [DEBUG] plugin: terraform: 	/usr/local/Cellar/go/1.7.3/libexec/src/reflect/value.go:434 +0x5c8
2016/11/15 16:59:34 [DEBUG] plugin: terraform: reflect.Value.Call(0xc42045b5c0, 0xc4201b4670, 0x13, 0xc420169ed0, 0x3, 0x3, 0x0, 0x0, 0x0)
2016/11/15 16:59:34 [DEBUG] plugin: terraform: 	/usr/local/Cellar/go/1.7.3/libexec/src/reflect/value.go:302 +0xa4
2016/11/15 16:59:34 [DEBUG] plugin: terraform: net/rpc.(*service).call(0xc42033ff00, 0xc42033fec0, 0xc420399ff0, 0xc42047cd80, 0xc4200c5fc0, 0x2364340, 0xc42070e9a0, 0x16, 0x2364380, 0xc42070ec30, ...)
2016/11/15 16:59:34 [DEBUG] plugin: terraform: 	/usr/local/Cellar/go/1.7.3/libexec/src/net/rpc/server.go:383 +0x148
2016/11/15 16:59:34 [DEBUG] plugin: terraform: created by net/rpc.(*Server).ServeCodec
2016/11/15 16:59:34 [DEBUG] plugin: terraform: 	/usr/local/Cellar/go/1.7.3/libexec/src/net/rpc/server.go:477 +0x421
2016/11/15 16:59:34 [ERROR] root: eval: *terraform.EvalReadDataApply, err: data.aws_acm_certificate.dns_zone_wildcard: unexpected EOF
2016/11/15 16:59:34 [ERROR] root: eval: *terraform.EvalSequence, err: data.aws_acm_certificate.dns_zone_wildcard: unexpected EOF
2016/11/15 16:59:34 [ERROR] root: eval: *terraform.EvalOpFilter, err: data.aws_acm_certificate.dns_zone_wildcard: unexpected EOF
2016/11/15 16:59:34 [ERROR] root: eval: *terraform.EvalSequence, err: data.aws_acm_certificate.dns_zone_wildcard: unexpected EOF
```

**Comment by jbardin** [comment]
I believe both these are fixed in the current releases, duplicate of #9989

**Comment by mrwacky42** [comment]
Oops, I stopped looking for solutions when I found this issue.
